### PR TITLE
♻️ refactor(eest): Migrate `env_init` to `eest make env`

### LINKED ISSUE #970: ♻️ cli(eest): Move environment config generation under `make` CLI
The `evm_init` CLI should be be a sub command of `et make`.  The `make` command namespace includes commands that generates project files.

Proposed CLI for generating env file: `uv run et make env`. 

_Originally posted by @raxhvl in https://github.com/ethereum/execution-spec-tests/issues/950#issuecomment-2488253410_

### PR DESCRIPTION
**Pull request by raxhvl**
## 🗒️ Description
This _tiny_ PR introduces the `eest  make config` command. It would be nice to have this included in the upcoming release; I'd rather not have a [dangling module](https://github.com/ethereum/execution-spec-tests/blob/ed433a7bd73c1e8ecdb5bbb5f48faf303c89e0cf/src/cli/eest/quotes.py) in it.

## 🔗 Related Issues

Closes #970 

## ✅ Checklist
- [x] All: Set appropriate labels for the changes.
- [x] All: Considered squashing commits to improve commit history.
- [x] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [x] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [x] Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been added to [converted-ethereum-tests.txt](/ethereum/execution-spec-tests/blob/main/converted-ethereum-tests.txt).
- [x] Tests: A PR with removal of converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been opened.
- [x] Tests: Included the type and version of evm t8n tool used to locally execute test cases:  e.g., ref with commit hash or geth 1.13.1-stable-3f40e65.
- [x] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

**Review by danceratopz — COMMENTED** [review]
Thanks @raxhvl! It would def be nice to have this cleaned up in the upcoming release.

I had in mind we'd name this `make` sub-comand `env`, i.e.,
```
uv run eest make env
```
as I think we're likely to subsequently add a `config` sub-command that helps users create a user-specific config profile for use within EEST.

Were you anticipating having the `env.vaml` generation just one step of creating a broader user config profile? Ideas welcome.

Btw, we're suffering from unrelated fails again in the CI - fix in progress.

File: docs/dev/configurations.md [hunk 1]
```diff
@@ -19,17 +19,17 @@ Application-wide [environment configuration](https://www.12factor.net/config), w
 
 This file will not be tracked by git, making it safe for storing local secrets.
 
-To get started, run the command [env_init](../library/cli/env_init.md) cli to initialize your environment configuration.
+To get started, run the command [eest make config](../library/cli/eest.md) cli to initialize your environment configuration.
```
Comment (danceratopz) [inline docs/dev/configurations.md:22]:
```suggestion
To get started, run the command [eest make env](../library/cli/eest.md) cli to initialize your environment configuration.
```

File: docs/dev/configurations.md [hunk 2]
```diff
@@ -19,17 +19,17 @@ Application-wide [environment configuration](https://www.12factor.net/config), w
...
 
 ### Usage
 
 #### 1. Generate env file
 
-Run the [`env_init`](../library/cli/env_init.md) cli tool.
+Run the [`eest make config`](../library/cli/eest.md) cli tool.
```
Comment (danceratopz) [inline docs/dev/configurations.md:28]:
```suggestion
Run the [`eest make env`](../library/cli/eest.md) cli tool.
```

File: docs/CHANGELOG.md [hunk 3]
```diff
@@ -58,6 +58,7 @@ Test fixtures for use by clients are available for each release on the [Github r
 - ✨ Add the `eest make test` command, an interactive CLI that helps users create a new test module and function ([#950](https://github.com/ethereum/execution-spec-tests/pull/950)).
 - ✨ Add the `eest clean` command that helps delete generated files and directories from the repository ([#980](https://github.com/ethereum/execution-spec-tests/pull/980)).
 - ✨ Add framework changes for EIP-7742, required for Prague devnet-5 ([#931](https://github.com/ethereum/execution-spec-tests/pull/931)).
+- ✨ Add the `eest make config` command that helps generate the default configuration file (env.yml)([#996](https://github.com/ethereum/execution-spec-tests/pull/996)).
```
Comment (danceratopz) [inline docs/CHANGELOG.md:61]:
```suggestion
- ✨ Add the `eest make env` command that generates a default environment config file (`env.yml`)([#996](https://github.com/ethereum/execution-spec-tests/pull/996)).
```

**Review by danceratopz — APPROVED** [review]
Thanks for the update @raxhvl, just fixed up a couple of other instances in the docs. Otherwise, looks good to me!

**Review by spencer-tb — APPROVED** [review]
LGTM! Nice addition :)

Just one question - was there a reason to opt for `env.yaml` over `.env`?
My gut would have said to use `.env` instead - and load the env vars directly with `loadenv()` etc

**Comment by danceratopz** [comment]
> Just one question - was there a reason to opt for `env.yaml` over `.env`?
> My gut would have said to use `.env` instead - and load the env vars directly with `loadenv()` etc

A `.env` normally contains key-value pairs, we're using a yaml file. :slightly_smiling_face:

**Comment by raxhvl** [comment]
> LGTM! Nice addition :)
> 
> Just one question - was there a reason to opt for `env.yaml` over `.env`? My gut would have said to use `.env` instead - and load the env vars directly with `loadenv()` etc

YAML is preferred over `.env` primarily because it supports data types, enabling the use of nested configurations. For example, consider the gentest config:
```yml
remote_nodes:
  - name: mainnet_archive
    # Replace with your Ethereum RPC node URL
    node_url: http://example.com/
    # Optional: Headers for RPC requests
    rpc_headers:
      client-secret: <secret>
```

**Pull request merged**